app/src/main/java/com/epfl/drawyourpath/challenge/TemporaryUser.kt

```diff
@@ -0,0 +1,111 @@
+package com.epfl.drawyourpath.challenge
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.*
+
+/**
+ * Temporary user which will need to be moved into user
+ *
+ * @param dailyGoals a doubly linked-List which should always have the most recent dailyGoal as first
+ * @param tournaments a list of tournaments the user takes part in
+ * @param trophies the list of trophies earned by the user (should not be initialized at creation)
+ */
+data class TemporaryUser(
+    private val dailyGoals: LinkedList<DailyGoal> = LinkedList(),
+    val tournaments: List<Tournament> = listOf(),
+    private val trophies: EnumMap<Trophy, LocalDate> = EnumMap(Trophy::class.java)
+) : java.io.Serializable {
+
+
+    /**
+     * get Today's Daily Goal or create a new one if there is no Daily Goals for today
+     *
+     * @return DailyGoal representing today's Daily Goal
+     */
+    fun getTodayDailyGoal(): DailyGoal {
+        if (dailyGoals.isEmpty()) {
+            dailyGoals.addFirst(DEFAULT_DAILY_GOAL)
+            return dailyGoals.first
+        }
+        if (dailyGoals.first.date == LocalDate.now()) {
+            return dailyGoals.first
+        }
+        val newDailyGoal = dailyGoals.first.createNewGoalFromThis()
+        dailyGoals.addFirst(newDailyGoal)
+        return newDailyGoal
+    }
+
+    /**
+     * function to call when a trophy is earned
+     * it will add the trophy along with the current date

```

Comment:
Maybe you can considerate the case when the user win the trophy more than one time. For example, a user can win the same trophy each time he made a run of 10 km(the most of the running do that for simplification)

Reply:
I  do not know exactly how the trophy system will work but it is easy to change as we could change the LocalDate into a list of LocalDate but the placeholder I used were supposed to be a total distance over the time instead of a one time run.

Reply:
OK, I understand, We will discuss during the next meeting